ngrinder-core/src/main/java/net/grinder/AgentController.java

```diff
@@ -95,12 +96,29 @@ public AgentController(Condition eventSyncCondition, AgentConfig agentConfig) {
 		this.m_agentControllerServerListener = new AgentControllerServerListener(m_eventSynchronization, LOGGER);
 		// Set it with the default name
 		this.m_agentIdentity = new AgentControllerIdentityImplementation(agentConfig.getAgentHostID(), agentConfig.getBroadcastIP());
-		this.m_agentIdentity.setRegion(agentConfig.getRegion());
+		this.m_agentIdentity.setSubregion(agentConfig.getSubregion());
+
+		String legacyOwner = getOwnerFromLegacyRegionConfig(agentConfig.getRegion());
+		String configuredOwner = agentConfig.getOwner();
+
+		m_agentIdentity.setOwner(legacyOwner);
+		if (isNotEmpty(configuredOwner)) {
+			this.m_agentIdentity.setOwner(configuredOwner);
+		}
```

Comment:
what about using `if - else`? and add `this` to first statement?

Reply:
It is Fixed